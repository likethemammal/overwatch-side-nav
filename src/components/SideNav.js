import React, { Component } from 'react'

import css from './SideNav.less'

import Item from './Item'
import Scroll from './Scroll'

import _ from 'underscore'

export default class SideNav extends Component {
    static defaultProps = {
        className: css.container,
        zIndex: 1000000,
    }

    state = {
        hash: window.location.hash,
    }

    onHashChange = (hash) => {
        this.setState({
            hash,
        })
    }

    render() {

        const { className, items, zIndex, ...rest } = this.props
        const ids = _.pluck(items, 'id')
        const { hash } = this.state
        const state = {
            hash,
            onHashChange: this.onHashChange,
        }
        const containerStyles = {
            zIndex,
        }

        return <div
            className={className}
            style={containerStyles}
        >
            <Scroll
                {...{ids}}
                {...state}
            />
            {items.map((item, i) => <Item
                key={i}
                {...state}
                {...rest}
                {...item}
            />)}
        </div>
    }
}