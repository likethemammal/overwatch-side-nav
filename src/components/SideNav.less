.container {
    position: fixed;
    right: 0;
    color: white;
    top: 50%;
    transform: translateY(-50%);
}