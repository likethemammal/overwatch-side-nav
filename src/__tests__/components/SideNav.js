import React from 'react';
import {shallow} from 'enzyme';

import general from '@likethemammal/general-components'

const shared = general.__tests__._shared.tests

import SideNav from '../../components/'

const items = [
    {
        label: 'Label',
        id: 'some_id',
        icon: [],
        Component: <div>test</div>,
    },
    {
        label: 'Test',
        id: 'slug',
        icon: [],
        Component: <div>The fuck</div>,
    },
]


describe('SideNav', () => {
    // Render a checkbox with label in the document
    const component = shallow(
        <SideNav
            items={items}
        />
    )

    shared.SNAPSHOTS_SHOULD_MATCH([
        component
    ])

})