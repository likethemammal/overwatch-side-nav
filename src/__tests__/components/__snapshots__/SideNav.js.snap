// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav Snapshot 0 should match 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideNav
    className={undefined}
    items={
      Array [
        Object {
          "Component": <div>
            test
          </div>,
          "icon": Array [],
          "id": "some_id",
          "label": "Label",
        },
        Object {
          "Component": <div>
            The fuck
          </div>,
          "icon": Array [],
          "id": "slug",
          "label": "Test",
        },
      ]
    }
    zIndex={1000000}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Scroll
          hash=""
          ids={
            Array [
              "some_id",
              "slug",
            ]
          }
          onHashChange={[Function]}
        />,
        Array [
          <Item
            Component={
              <div>
                test
              </div>
            }
            activeColor="#00c3ff"
            className={undefined}
            defaultColor="#323c50"
            hash=""
            icon={Array []}
            id="some_id"
            label="Label"
            onHashChange={[Function]}
            textColor="white"
          />,
          <Item
            Component={
              <div>
                The fuck
              </div>
            }
            activeColor="#00c3ff"
            className={undefined}
            defaultColor="#323c50"
            hash=""
            icon={Array []}
            id="slug"
            label="Test"
            onHashChange={[Function]}
            textColor="white"
          />,
        ],
      ],
      "className": undefined,
      "style": Object {
        "zIndex": 1000000,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "hash": "",
          "ids": Array [
            "some_id",
            "slug",
          ],
          "onHashChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "Component": <div>
            test
          </div>,
          "activeColor": "#00c3ff",
          "className": undefined,
          "defaultColor": "#323c50",
          "hash": "",
          "icon": Array [],
          "id": "some_id",
          "label": "Label",
          "onHashChange": [Function],
          "textColor": "white",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "Component": <div>
            The fuck
          </div>,
          "activeColor": "#00c3ff",
          "className": undefined,
          "defaultColor": "#323c50",
          "hash": "",
          "icon": Array [],
          "id": "slug",
          "label": "Test",
          "onHashChange": [Function],
          "textColor": "white",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Scroll
            hash=""
            ids={
              Array [
                "some_id",
                "slug",
              ]
            }
            onHashChange={[Function]}
          />,
          Array [
            <Item
              Component={
                <div>
                  test
                </div>
              }
              activeColor="#00c3ff"
              className={undefined}
              defaultColor="#323c50"
              hash=""
              icon={Array []}
              id="some_id"
              label="Label"
              onHashChange={[Function]}
              textColor="white"
            />,
            <Item
              Component={
                <div>
                  The fuck
                </div>
              }
              activeColor="#00c3ff"
              className={undefined}
              defaultColor="#323c50"
              hash=""
              icon={Array []}
              id="slug"
              label="Test"
              onHashChange={[Function]}
              textColor="white"
            />,
          ],
        ],
        "className": undefined,
        "style": Object {
          "zIndex": 1000000,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "hash": "",
            "ids": Array [
              "some_id",
              "slug",
            ],
            "onHashChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "Component": <div>
              test
            </div>,
            "activeColor": "#00c3ff",
            "className": undefined,
            "defaultColor": "#323c50",
            "hash": "",
            "icon": Array [],
            "id": "some_id",
            "label": "Label",
            "onHashChange": [Function],
            "textColor": "white",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "Component": <div>
              The fuck
            </div>,
            "activeColor": "#00c3ff",
            "className": undefined,
            "defaultColor": "#323c50",
            "hash": "",
            "icon": Array [],
            "id": "slug",
            "label": "Test",
            "onHashChange": [Function],
            "textColor": "white",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
